src/map/script.c

```diff
@@ -15547,34 +15547,47 @@ static BUILDIN(recovery)
 }
 
 /*==========================================
- * Get your pet info: getpetinfo(n)
- * n -> 0:pet_id 1:pet_class 2:pet_name
- * 3:friendly 4:hungry, 5: rename flag.
+ * Get your pet info: getpetinfo
  *------------------------------------------*/
 static BUILDIN(getpetinfo)
 {
 	struct map_session_data *sd = script->rid2sd(st);
-	struct pet_data *pd;
-	int type=script_getnum(st,2);
+	if (sd == NULL)
+		return true;
 
-	if (sd == NULL || sd->pd == NULL) {
-		if (type == 2)
-			script_pushconststr(st,"null");
+	struct pet_data *pd = sd->pd;
+	int type = script_getnum(st, 2);
```

Comment:
@4144 @MishimaHaruna what is criteria for this?
I see at some commits these are moved to where it is used and for most other things, it's declared at start of function/block

Reply:
In past vars must be moved to top, because outdated visual studio, It not supported any modeern C standards (i think it was vs 2002)

But now we not using this outdated version.
And for good style variables must be defines as close as possible to variables usage. Also vars must use smaller scope as possible.
This is simple rules to bypass different typos and other kind errors.

This mean comment from @hemagx is invalid here.